pkg/cloudprovider/vsphere/loadbalancer/access.go

```diff
@@ -0,0 +1,494 @@
+		if checkTags(item.Tags, tags...) {
+			return &item, nil
+		}
+	}
+	return nil, nil
+}
+
+func (a *access) UpdateLoadBalancerService(lbService *model.LBService) error {
+	_, err := a.broker.UpdateLoadBalancerService(*lbService)
+	if err != nil {
+		return errors.Wrapf(err, "updating load balancer service %s (%s) failed", *lbService.DisplayName, *lbService.Id)
+	}
+	return nil
+}
+
+func (a *access) DeleteLoadBalancerService(id string) error {
+	err := a.broker.DeleteLoadBalancerService(id)
+	if isNotFoundError(err) {
+		return nil
+	}
+	if err != nil {
+		return errors.Wrapf(err, "deleting load balancer service %s failed", id)
+	}
+	return nil
+}
+
+func (a *access) findAppProfilePathByName(profileName string, resourceType string) (string, error) {
+	list, err := a.broker.ListAppProfiles()
+	if err != nil {
+		return "", err
+	}
+	path := ""
+	for _, item := range list {
+		itemResourceType, err := item.String("resource_type")
+		if err != nil {
+			return "", errors.Wrapf(err, "findAppProfilePathByName cannot find field resource_type")
+		}
+		itemName, err := item.String("display_name")
+		if err != nil {
+			return "", errors.Wrapf(err, "findAppProfilePathByName cannot find field name")
+		}
+		if itemResourceType == resourceType && itemName == profileName {
+			if path != "" {
+				return "", fmt.Errorf("profile name %s for resource type %s is not unique", profileName, resourceType)
+			}
+			path, err = item.String("path")
+			if err != nil {
+				return "", errors.Wrapf(err, "findAppProfilePathByName cannot find field path")
+			}
+		}
+	}
+	if path == "" {
+		return "", fmt.Errorf("application profile named %s of type %s not found", profileName, resourceType)
+	}
+	return path, nil
+}
+
+func (a *access) GetAppProfilePath(class LBClass, protocol corev1.Protocol) (string, error) {
+	profileReference, err := class.AppProfile(protocol)
+	if err != nil {
+		return "", err
+	}
+	if profileReference.Identifier != "" {
+		return profileReference.Identifier, nil
+	}
+	resourceType := ""
+	switch protocol {
+	case corev1.ProtocolTCP:
+		resourceType = model.LBAppProfile_RESOURCE_TYPE_LBFASTTCPPROFILE
+	case corev1.ProtocolUDP:
+		resourceType = model.LBAppProfile_RESOURCE_TYPE_LBFASTUDPPROFILE
+	default:
+		return "", fmt.Errorf("Unsupported protocol %s", protocol)
+	}
+	return a.findAppProfilePathByName(profileReference.Name, resourceType)
+}
+
+func (a *access) CreateVirtualServer(clusterName string, objectName types.NamespacedName, class LBClass, ipAddress string,
+	mapping Mapping, lbServicePath, applicationProfilePath string, poolPath *string) (*model.LBVirtualServer, error) {
+	allTags := append(class.Tags(), clusterTag(clusterName), serviceTag(objectName), portTag(mapping))
+	virtualServer := model.LBVirtualServer{
+		Description: strptr(fmt.Sprintf("virtual server for cluster %s, service %s created by %s",
+			clusterName, objectName, AppName)),
+		DisplayName:            strptr(fmt.Sprintf("cluster:%s:%s", clusterName, objectName)),
```

Comment:
The display name format should be a constant or generated from one common function.

Reply:
Seems like we can use `GetLoadBalancerName()`

Reply:
Introduced some display name functions in a central place for various combinations.
the `access` structure should not know anything about the `lbProvider` struct. Therefore we cannot use a method from this unknown structure. Instead we just use a common generic function.